lib/utils/common.py

```diff
@@ -1084,21 +1082,23 @@ def _manually_compute_license_data_size(
                 host_record_overhead = 39
 
             host_unique_data = host_memory_bytes + host_data_bytes
-            ns_unique_data += host_unique_data
-            ns_record_overhead += host_master_objects * host_record_overhead
-            ns_master_objects += host_master_objects
-
-        if version.LooseVersion(constants.SERVER_NO_BYTE_OVERHEAD_FIRST_VERSION) <= version.LooseVersion(host_build_version):
-            # For 8.0 and above - don't subtract overhead
-            ns_unique_data = round((ns_unique_data / ns_repl_factor))
-        else:
-             # For versions below 8.0 - subtract overhead
-            ns_unique_data = round((ns_unique_data / ns_repl_factor) - ns_record_overhead)
+            
+            # Calculate host contribution based on its individual version
+            if version.LooseVersion(constants.SERVER_NO_BYTE_OVERHEAD_FIRST_VERSION) <= version.LooseVersion(host_build_version):
+                # For 8.0 and above - don't subtract overhead for this host
+                host_license_contribution = host_unique_data / ns_repl_factor
+            else:
+                # For versions below 8.0 - subtract overhead for this host
+                host_license_contribution = (host_unique_data / ns_repl_factor) - (host_master_objects * host_record_overhead)
+            
+            ns_unique_data += host_license_contribution
         
+        # Clamp namespace license data to ensure it's never negative
+        ns_unique_data_clamped = max(0, ns_unique_data)
```

Comment:
If the ns_unique_data is negative, issuing a warning or debug log would be good.